### PR TITLE
Migrated tests to mock clock

### LINKED ISSUE #106989: endpointslice_controller_test.go is using time.Sleep
In https://github.com/kubernetes/kubernetes/pull/88745 I added a unit tests that are using time.Sleep. It's bad: it makes our tests slower than they need to be and involves some risk of races.

We should migrate this test to mocked clock.

This requires implementing a new workqueue.NewNamedRateLimitingQueueWitthCustomClock, similarly as workload.NewDelayingQueueWithCustomClock and migrating TestPodAddsBatching and TestPodUpdatesBatching to use that.

/kind cleanup

### PR DESCRIPTION
**Pull request by NikhilSharmaWe**
#### What type of PR is this?
Fixing a bug 

#### What this PR does / why we need it:
This PR migrates functions `TestPodAddsBatching` and `TestPodUpdatesBatching` to use mock clock.
#### Which issue(s) this PR fixes:

Fixes #106989 

#### Does this PR introduce a user-facing change?

```release-note
None
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by NikhilSharmaWe** [comment]
@mborz 
- What package should be imported for migrating mock clock ? I have imported package `github.com/benbjohnson/clock`, but it is showing an error -
`pkg/controller/endpointslice/endpointslice_controller_test.go:27:2: cannot find package "github.com/benbjohnson/clock" in any of:
	/home/prow/go/src/k8s.io/kubernetes/vendor/github.com/benbjohnson/clock (vendor tree)
	/usr/local/go/src/github.com/benbjohnson/clock (from $GOROOT)
	/home/prow/go/src/github.com/benbjohnson/clock (from $GOPATH)
make[1]: *** [Makefile:365: vet] Error 1`.

- Could you guide about what packages should be imported here, what improvements and how to proceed in this PR.

**Comment by NikhilSharmaWe** [comment]
@mborsz
- What package should be imported for migrating mock clock ? I have imported package `github.com/benbjohnson/clock`, but it is showing an error -
`pkg/controller/endpointslice/endpointslice_controller_test.go:27:2: cannot find package "github.com/benbjohnson/clock" in any of:
	/home/prow/go/src/k8s.io/kubernetes/vendor/github.com/benbjohnson/clock (vendor tree)
	/usr/local/go/src/github.com/benbjohnson/clock (from $GOROOT)
	/home/prow/go/src/github.com/benbjohnson/clock (from $GOPATH)
make[1]: *** [Makefile:365: vet] Error 1`.

- Could you guide about what packages should be imported here, what improvements and how to proceed in this PR.

**Comment by NikhilSharmaWe** [comment]
@mborsz

What package should be imported for migrating mock clock ? I have imported package github.com/benbjohnson/clock, but it is showing an error -
pkg/controller/endpointslice/endpointslice_controller_test.go:27:2: cannot find package "github.com/benbjohnson/clock" in any of: /home/prow/go/src/k8s.io/kubernetes/vendor/github.com/benbjohnson/clock (vendor tree) /usr/local/go/src/github.com/benbjohnson/clock (from $GOROOT) /home/prow/go/src/github.com/benbjohnson/clock (from $GOPATH) make[1]: *** [Makefile:365: vet] Error 1.

Could you guide about what packages should be imported here, what improvements and how to proceed in this PR.

**Comment by dims** [comment]
@NikhilSharmaWe you should probably ask sig-scheduling for help (slack or email or meeting). In general if you want to use a new package, you need to add the dependency as shown here https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/godep.md#adding-a-dep

**Comment by NikhilSharmaWe** [comment]
I am getting an error when I run `hack/run-in-gopath.sh go get -d github.com/benbjohnson/clock` for adding package `github.com/benbjohnson/clock`.
```
ln: /e/githubRepos/kubernetes/_output/local/go/src/k8s.io/kubernetes/kubernetes: cannot overwrite directory
!!! [0106 20:17:28] Call tree:
!!! [0106 20:17:28]  1: /e/githubRepos/kubernetes/hack/lib/golang.sh:509 kube::golang::create_gopath_tree(...)
!!! [0106 20:17:28]  2: hack/run-in-gopath.sh:31 kube::golang::setup_env(...)

```
@mborsz Could you please guide me about how can I add this dependency successfully ?

**Comment by dims** [comment]
This PR has needs a kind/ label. Please pick one from https://github.com/kubernetes/kubernetes/labels?q=kind and apply it. Please see https://prow.k8s.io/command-help on how to use the bot to apply labels

File: pkg/controller/endpointslice/endpointslice_controller_test.go [hunk 1]
```diff
@@ -24,6 +24,7 @@ import (
 	"testing"
 	"time"
 
+	"github.com/benbjohnson/clock"
```
Comment (danwinship) [inline pkg/controller/endpointslice/endpointslice_controller_test.go:27]: This dependency does not currently exist in the kubernetes tree... is this PR part of some larger project? Does it depend on some other PR?

**Pull request closed without merging**